### ISSUE TITLE
Error while parsing stream redirection inside dollar-parentheses

### ISSUE THREAD

**Issue by Yanpas** [tree-sitter/tree-sitter-bash#34]
Snippet which fails:
```
function ff () {
    local xxxx="$(<aa)"
    case x in
      * ) echo "blabla"; return 1 ;;
    esac &> foo || { echo "ffff"; return 0; }
}
```

**Comment by maxbrunsfeld** [comment]
What does `$(<aa)` mean? I'm not familiar with that syntax.

**Comment by Yanpas** [comment]
it means store the content of a file `aa` in variable `xxxx`. Here is example:
```bash
function ff () {
    local xxxx="$(<aa)"
    echo "xxxx is $xxxx"
}

echo foo > aa

ff
```
output is `xxxx is foo`

**Comment by maxbrunsfeld** [comment]
Ah, found a stackoverflow question about this: https://stackoverflow.com/questions/40194222/why-does-input-redirection-work-differently-when-in-command-substitution. Seems like a special construct just within command substitution?

**Comment by Yanpas** [comment]
Did you mean [process substitution](https://www.gnu.org/software/bash/manual/html_node/Process-Substitution.html)? It has slightly different syntax.

The syntax above is analogous to `$(cat aa)`

**Comment by maxbrunsfeld** [comment]
No, I just meant that people seemed to refer to that `$(<x)` construct as a special shorthand command substitution, basically equivalent to `$(cat x)`. I realize that there's actually no command present though. Do you know of any docs for that syntax, and if it has a name?

**Comment by Yanpas** [comment]
According to official docs it is short form of cat command substitution:

https://www.gnu.org/software/bash/manual/bash.html#Command-Substitution

**Comment by maxbrunsfeld** [comment]
Perfect; thanks so much for finding that.

**Comment by maxbrunsfeld** [comment]
The fix is out in v0.13.5.

**Comment by Yanpas** [comment]
Thanks!
